File: CS/WinWebSolution.Module/Updater.cs
using System;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            int i;
            Department[] department = new Department[3];
            string[] departmentTitles = new string[] {
                "Development Department",
                "QA Department",
                "Sales Department"
            };
            for (i = 0; i <=2; i++)
                {
                department[i] = Session.FindObject<Department>(new BinaryOperator("Title", department[i]));
                    if(department[i] == null) {
                    department[i] = new Department(Session);
                    department[i].Title = departmentTitles[i];
                    department[i].Office = (101 + i).ToString();
                    department[i].Save();
                }
            }
            SimpleUser user = Session.FindObject<SimpleUser>(new BinaryOperator("UserName", "Sam"));
            if(user == null) {
                user = new SimpleUser(Session);
                user.UserName = "Sam";
                user.FullName = "";
            }
            user.IsAdministrator = true;
            user.SetPassword("");
            user.Save();
        }
    }
}

File: VB/WinWebSolution.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
    Public Class Updater
        Inherits ModuleUpdater
        Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
            MyBase.New(session, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            Dim i As Integer
            Dim department(2) As Department
            Dim departmentTitles() As String = { "Development Department", "QA Department", "Sales Department" }
            For i = 0 To 2
                department(i) = Session.FindObject(Of Department)(New BinaryOperator("Title", department(i)))
                    If department(i) Is Nothing Then
                    department(i) = New Department(Session)
                    department(i).Title = departmentTitles(i)
                    department(i).Office = (101 + i).ToString()
                    department(i).Save()
                    End If
            Next i
            Dim user As SimpleUser = Session.FindObject(Of SimpleUser)(New BinaryOperator("UserName", "Sam"))
            If user Is Nothing Then
                user = New SimpleUser(Session)
                user.UserName = "Sam"
                user.FullName = ""
            End If
            user.IsAdministrator = True
            user.SetPassword("")
            user.Save()
        End Sub
    End Class
End Namespace
